File: d08.jl
day = "08"
lines = open("inputs/$(day).txt") do f
    readlines(f); #
end

lines = [parse(Int, x) for x in split(lines[1], " ")]

function parseTree(T)
    nc, nm = T[1:2]
    T = T[3:end]

    totals = 0
    values = []
    for i in 1:nc
        total, value, T = parseTree(T)
        totals += total
        push!(values, value)
    end

    totals += sum(T[1:nm])
    if nc == 0
        return totals, totals, T[nm+1:end]
    else
        value = sum([values[j] for j in T[1:nm]
                     if ((j > 0) & (j <= length(values)))])
        return totals, value, T[nm+1:end]
    end
end

t, v, r = parseTree(lines)
println("Day ", day)
println("-------------")
println("Part 1: ", t)
println("Part 2: ", v)
println("remaining:", r)

File: d08.py
from util import read
import numpy as np

day = '08'
print(f'Day {day}')
print('------')

x = read(day)
x = [int(y) for y in x[0].split(' ')]

def parse(data):
    nc, nm = data[:2]
    data = data[2:]
    totals = 0
    values = []

    for i in range(nc):
        total, value, data = parse(data)
        totals += total
        values.append(value)

    totals += sum(data[:nm])

    if nc == 0:
        print(nc, nm, len(data))
        return (totals, totals, data[nm:])
    else:
        value = sum([values[j-1] for j in data[:nm]
                     if j > 0 and j <= len(values)])
        return (totals, value, data[nm:])

total, value, remaining = parse(x)
print("Part 1:", total)
print("Part 2:", value)
